### PR TITLE
fix: [CDS-69814]: modified service api call to take yaml

### PR DESCRIPTION
**Pull request by sahithikolichala**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by tan-nhu — APPROVED** [review]

**Review by cjlee01 — APPROVED** [review]

File: src/modules/75-cd/components/PipelineSteps/DeployServiceStep/NewEditServiceModal.tsx [hunk 1]
```diff
@@ -64,7 +64,12 @@ const cleanData = (values: ServiceRequestDTO): ServiceRequestDTO => {
     orgIdentifier: values.orgIdentifier,
     projectIdentifier: values.projectIdentifier,
     description: newDescription,
-    tags: values.tags
+    tags: values.tags,
+    yaml: JSON.stringify({
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/PipelineSteps/DeployServiceStep/NewEditServiceModal.tsx:68]: `JSON.stringify` will not convert the value to `yaml`. Use `yamlStringify` from `src/modules/10-common/utils/YamlHelperMethods.ts`

**Review by pankajsingh-harness — CHANGES_REQUESTED** [review]
JSON.stringify will not convert the value to yaml. Use yamlStringify from src/modules/10-common/utils/YamlHelperMethods.ts

**Review by pankajsingh-harness — APPROVED** [review]

**Comment by sahithikolichala** [comment]
retrigger jest

**Comment by sahithikolichala** [comment]
retrigger jest

**Pull request merged**